### ISSUE TITLE
Update alerting API to return errutil errors

### ISSUE THREAD

**Issue by yuri-tceretian** [grafana/grafana#53136]
PR #47504 introduced general recommendations for error handling as well as a tiny framework that would allow providing more conscious errors to different parties (end-user, administrators, developers).
The alerting API needs to be updated to follow the new [guidelines ](https://github.com/grafana/grafana/blob/main/contribute/architecture/backend/errors.md#declaring-errors).

**Comment by knownasaritra** [comment]
is this still open ?

**Comment by yuri-tceretian** [comment]
@knownasaritra, yes. The work has not been planned yet.

**Comment by knownasaritra** [comment]
if possible can I work on this ?

**Comment by joel113** [comment]
I drafted an implementation as an excercise for myself and as I am using Grafana lately quite much. I did enjoy the design of the new error type `github.com/grafana/grafana/pkg/util/errutil/errors.go`. I did an implementation which uses the new error type `github.com/grafana/grafana/pkg/util/errutil/errors.go` in the alerting API implementation `github.com/grafana/grafana/pkg/api/alerting.go` while considering the the `Err` method of `github.com/grafana/grafana/pkg/api/response/response.go` and `github.com/grafana/grafana/pkg/util/errutil/status.go`. When drafting the implementation I found some http status codes in `alerting.go` which seem to have no representation in `status.go`, e.g. `HTTP 422` in [line 178 of alerting.go](https://github.com/grafana/grafana/blob/main/pkg/api/alerting.go#L178)

```
return response.Error(422, validationErr.Error(), nil)
```

Would you rather turn this error code in something which is available in `status.go` or would you extend `status.go` or do I am completely wrong here?

Is there anything else to do than to adapt `alerting.go`? Are there any tests to adapt? Or am I completely wrong here?

**Comment by yuri-tceretian** [comment]
All contributions are highly appreciated and welcome! Please open a PR and I will gladly review it!

@joel113 thanks for looking into this but the file you are referring to belongs to a legacy API, which will be removed in the next major version (we have plans for Grafana 10). I would appreciate it if you could look into APIs of a new Grafana alerting https://github.com/grafana/grafana/tree/main/pkg/services/ngalert/api.
However, if you have changes in legacy code we can review it too so they won't be lost :)

As for missing status, the list https://github.com/grafana/grafana/blob/main/pkg/util/errutil/status.go  does not seem comprehensive. I think this is fine to add missing statuses to that list.

**Comment by omarghetti** [comment]
is this still open to work on?

**Comment by joel113** [comment]
@omarghetti I am not sure if I am the right person to answer whether this is still open. I can say, that I did start working on this as part of my excercise to look into grafana. Unfortunately, I have usually little time and in the upcoming weeks no time at all. Hence, I will not provide a pr before 11/2022. Therefore, it would be no pitty at all if you would start working. So far I went through the ngalerting codebase and started to introduce the errutil error handling. In my opinion, the ngalerting codebase is not as easy and not as straightforward as the grafana alerting codebase which I investigated earlier. However, I think it is a nice task and so far I enjoyed it very much.
